agent/agent.go

```diff
@@ -373,9 +364,26 @@ func runChecker(ctx context.Context, checker health.Checker, probeCh chan<- heal
 
 	log.Debugf("Running checker %q.", checker.Name())
 
-	var probes health.Probes
-	checker.Check(ctx, &probes)
-	probeCh <- probes
+	checkResult := make(chan health.Probes)
+	go func() {
+		var probes health.Probes
+		checker.Check(ctx, &probes)
+		checkResult <- probes
```

Comment:
This, too, can block indefinitely and will result in a goroutine leak.

Reply:
If we can't rely on `checker.Check` to return after `ctx` is canceled, is it possible to avoid a goroutine leak or a blocked call? `Checker` has no methods to close or interrupt the on going health checks.